### PR TITLE
Add a new module for Claims

### PR DESCRIPTION
**Pull request by SatishBV**
## [APP-1369]

- Move claims stuff from Home to a new Claims framework.
- Create a separate store for Claims
- Handle navigation between claims and other parts of the app through Journeys

## Checklist

- [x] Dark/light mode verification
- [x] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

[APP-1366]: https://hedvig.atlassian.net/browse/APP-1366?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-1367]: https://hedvig.atlassian.net/browse/APP-1367?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-1368]: https://hedvig.atlassian.net/browse/APP-1368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-1369]: https://hedvig.atlassian.net/browse/APP-1369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Projects/Home/Sources/Viewables/ActiveSection.swift [hunk 1]
```diff
@@ -15,14 +16,12 @@ extension ActiveSection: Presentable {
...
 
-        let store: HomeStore = self.get()
-
         section.dynamicStyle = .brandGrouped(
             insets: .init(top: 0, left: 14, bottom: 0, right: 14),
             separatorType: .none
         )
 
-        let claims = ClaimSectionLoading()
+        let claims = Claims()
```
Comment (SatishBV) [inline Projects/Home/Sources/Viewables/ActiveSection.swift:24]: @sampettersson To display the claim cards stuff on home, I had to import Claims here.
Do you think there is a way by which I can bypass this import and call it via Ugglan instead?

File: Projects/Claims/Example/Sources/Debug.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import Apollo
+import ExampleUtil
+import Flow
+import Form
+import Foundation
+import Home
```
Comment (sampettersson) [inline Projects/Claims/Example/Sources/Debug.swift:6]: Does Claims depend on Home? We should avoid those cross-dependencies IMO!

File: Projects/Claims/Example/Sources/Debug.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            .onValue { presentHome(.makePending(switchable: true)) }
+
+        bag += section.appendRow(title: "Home - Pending non switchable")
+            .append(hCoreUIAssets.chevronRight.image)
+            .onValue { presentHome(.makePending(switchable: false)) }
+
+        bag += section.appendRow(title: "Home - With payment card").append(hCoreUIAssets.chevronRight.image)
+            .onValue { presentHome(combineMultiple([.makeActive(), .makePayInMethodStatus(.needsSetup)])) }
+
+        bag += section.appendRow(title: "Renewals - One renewal").append(hCoreUIAssets.chevronRight.image)
```
Comment (sampettersson) [inline Projects/Claims/Example/Sources/Debug.swift:70]: Did you forget to remove these?

File: Projects/Claims/Example/Sources/SceneDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        let tapGestureRecognizer = UITapGestureRecognizer()
+        tapGestureRecognizer.numberOfTouchesRequired = 3
+
+        bag += tapGestureRecognizer.signal(forState: .recognized)
+            .onValue { _ in self.bag.dispose()
+                self.run()
+            }
+
+        window?.addGestureRecognizer(tapGestureRecognizer)
```
Comment (sampettersson) [inline Projects/Claims/Example/Sources/SceneDelegate.swift:29]: I don't think you need all these, these where kind of a hack as Home couldn't be presented in a NavigationController correctly 😓

File: Projects/Home/Sources/Viewables/ActiveSection.swift (hunk 1 above)
Comment (sampettersson) [reply Projects/Home/Sources/Viewables/ActiveSection.swift:24]: @SatishBV Make home take a generic or just an AnyView, and then passing it in from ugglan should work!

**Review by sampettersson — CHANGES_REQUESTED** [review]
Great work!!

File: Projects/Home/Sources/Home.swift [hunk 5]
```diff
@@ -10,18 +10,21 @@ import hCore
 import hCoreUI
 import hGraphQL
 
-public struct Home {
+public struct Home<ClaimsContent: UIView> {
```
Comment (sampettersson) [inline Projects/Home/Sources/Home.swift:13]: Imo don't pass the whole hosting view here, just pass the SwiftUI view.

**Review by sampettersson — CHANGES_REQUESTED** [review]
Looks great!

**Pull request merged**